src/measure_theory/interval_integral.lean

```diff
@@ -209,66 +276,778 @@ lemma integral_add_adjacent_intervals (hfm : measurable f) (hab : interval_integ
+lemma integral_has_deriv_at_right (hfm : measurable f)
+  (hfi : interval_integrable f volume a b) (hb : continuous_at f b) :
+  has_deriv_at (λ u, ∫ x in a..u, f x) (f b) b :=
+(integral_has_strict_deriv_at_right hfm hfi hb).has_deriv_at
+
+/-- Fundamental theorem of calculus: if `f : ℝ → E` is integrable on `a..b` and `f` has a finite
+limit `c` almost surely at `b`, then the derivative of `u ↦ ∫ x in a..u, f x` at `b` equals `c`. -/
+lemma deriv_integral_of_tendsto_ae_right (hfm : measurable f)
+  (hfi : interval_integrable f volume a b) (hb : tendsto f (𝓝 b ⊓ volume.ae) (𝓝 c)) :
+  deriv (λ u, ∫ x in a..u, f x) b = c :=
+(integral_has_deriv_at_of_tendsto_ae_right hfm hfi hb).deriv
+
+/-- Fundamental theorem of calculus: if `f : ℝ → E` is integrable on `a..b` and `f` is continuous
+at `b`, then the derivative of `u ↦ ∫ x in a..u, f x` at `b` equals `f b`. -/
+lemma deriv_integral_right (hfm : measurable f) (hfi : interval_integrable f volume a b)
+  (hb : continuous_at f b) :
+  deriv (λ u, ∫ x in a..u, f x) b = f b :=
+(integral_has_deriv_at_right hfm hfi hb).deriv
+
+/-- Fundamental theorem of calculus-1: if `f : ℝ → E` is integrable on `a..b` and `f x` has a finite
+limit `c` almost surely at `a`, then `u ↦ ∫ x in u..b, f x` has derivative `-c` at `a`. -/
+lemma integral_has_deriv_at_of_tendsto_ae_left (hfm : measurable f)
+  (hfi : interval_integrable f volume a b) (ha : tendsto f (𝓝 a ⊓ volume.ae) (𝓝 c)) :
+  has_deriv_at (λ u, ∫ x in u..b, f x) (-c) a :=
+(integral_has_strict_deriv_at_of_tendsto_ae_left hfm hfi ha).has_deriv_at
+
+/-- Fundamental theorem of calculus-1: if `f : ℝ → E` is integrable on `a..b` and `f` is continuous
+at `a`, then `u ↦ ∫ x in u..b, f x` has derivative `-f a` at `a`. -/
+lemma integral_has_deriv_at_left (hfm : measurable f) (hfi : interval_integrable f volume a b)
+  (ha : continuous_at f a) :
+  has_deriv_at (λ u, ∫ x in u..b, f x) (-f a) a :=
+(integral_has_strict_deriv_at_left hfm hfi ha).has_deriv_at
+
+/-- Fundamental theorem of calculus: if `f : ℝ → E` is integrable on `a..b` and `f` has a finite
+limit `c` almost surely at `a`, then the derivative of `u ↦ ∫ x in u..b, f x` at `a` equals `-c`. -/
+lemma deriv_integral_of_tendsto_ae_left (hfm : measurable f)
+  (hfi : interval_integrable f volume a b) (hb : tendsto f (𝓝 a ⊓ volume.ae) (𝓝 c)) :
+  deriv (λ u, ∫ x in u..b, f x) a = -c :=
+(integral_has_deriv_at_of_tendsto_ae_left hfm hfi hb).deriv
+
+/-- Fundamental theorem of calculus: if `f : ℝ → E` is integrable on `a..b` and `f` is continuous
+at `a`, then the derivative of `u ↦ ∫ x in u..b, f x` at `a` equals `-f a`. -/
+lemma deriv_integral_left (hfm : measurable f) (hfi : interval_integrable f volume a b)
+  (hb : continuous_at f a) :
+  deriv (λ u, ∫ x in u..b, f x) a = -f a :=
+(integral_has_deriv_at_left hfm hfi hb).deriv
+
+/-!
+#### One-sided derivatives
+-/
+
+/-- Let `f` be a measurable function integable on `a..b`. The function `(u, v) ↦ ∫ x in u..v, f x`
```

Comment:
`integable` should be `integrable`, here and at several other places in the file.